source/common/http/http2/codec_impl_legacy.cc

```diff
@@ -333,6 +334,17 @@ void ConnectionImpl::StreamImpl::saveHeader(HeaderString&& name, HeaderString&&
 }
 
 void ConnectionImpl::StreamImpl::submitTrailers(const HeaderMap& trailers) {
+  const bool skip_encoding_empty_trailers =
+      trailers.empty() && parent_.skip_encoding_empty_trailers_;
+  if (skip_encoding_empty_trailers) {
```

Comment:
can you ASSERT(local_end_stream_) above this line? I think it should always be true?

Reply:
Added in 3f2c2d8.